.github/workflows/ci.yml

```diff
@@ -27,6 +27,8 @@ jobs:
       - run: cargo check --no-default-features
       - run: cargo check --no-default-features --features "use_alloc"
       - run: cargo test
+      - run: RUSTDOCFLAGS="-Dwarnings" cargo doc --all-features
+      - run: cargo clippy --all -- -Dwarnings
```

Comment:
Yes, it would be rejected. Same goes for `cargo doc` (and `cargo fmt` in the future as well).

Isn't that the point of having a CI, that the code is equally good (or bad)?

Reply:
# tl;dr

I'd like to have both a low bar for (especially new) contributors and try to acchieve a certain quality in the code base. Thus:
* I'm in favor of rejecting wrong documentation, as the result is user-facing and should be correct.
* I'm unsure if rejecting PRs with lint problems is a good idea.
* I'm very skeptical to reject PRs because of formatting.

## cargo clippy

I assume the codebase will quickly converge to a lint-free state (especially after your changes are merged). Once in that state, avoiding new lint problems should be easy.

From my personal experience, I found it demotivating if a PR is rejected because of a clippy-false-positive (or one that does not matter in the circumstances at hand). I'd like to avoid this situation as far as possible. 

* For one-off contributors, requesting and waiting for changes might be just as much effort as just merging and fixing the lint, possibly indicating that `clippy` should only warn instead of reject the PR.
* I assume frequent contributors do not introduce many new lints. (They may have clippy running on their local machines, so are aware of lint problems from the start.)

Generally, I think I'd prefer to only issue a warning (thus requiring human interaction to resolve it) instead of plainly rejecting the PR, assuming that it does not generate too much work on the contributors and the maintainers. I would be open to have `clippy` reject PRs if it's easier/better to implement/maintain/work with/others really want it/etc, as it sometimes gives really useful advice.

## cargo fmt

I guess the main advantage of having a `cargo fmt`d code base is that contributors have an idea how our code is written. However, for this it should be enough to "`cargo fmt` from time to time" (read: after a successfully merged PR). As such, I'd prefer *not* to reject PRs because of formatting problems (again to avoid frustration) - and instead have the formatting problems fixed automatically by our CI.